oauth2/src/main/java/com/okta/spring/boot/oauth/env/OktaOAuth2PropertiesMappingEnvironmentPostProcessor.java

```diff
@@ -175,6 +177,9 @@ private ConditionalMapPropertySource(String name, Map<String, Object> source, En
 
         @Override
         public Object getProperty(String name) {
+            if (OKTA_ISSUER_URI.equals(name) && OKTA_STATIC_DISCOVERY.equals(this.getName()) && !containsProperty(name)) {
+                throw new IllegalArgumentException("Mandatory property `" + OKTA_OAUTH_ISSUER + "` is missing");
```

Comment:
See my previous comment

Reply:
@bdemers 
Unfortunately approach with validators does't work for this property.
We can validate this property: we can define some pattern describing issuer URL, for example. And this case will work perfectly, but only if the issuer property is present.
If issuer-url is absent, then the same exception will be thrown:
`org.springframework.security.oauth2.client.registration.ClientRegistration$Builder.validateAuthorizationCodeGrantType -> java.lang.IllegalArgumentException: tokenUri cannot be empty`
This validation comes before all others.
I’ll continue looking for a better solution...

Reply:
I'm not sure I understand.
Taking a step back, if you run a similar example without the Okta Spring Boot starter (with only Spring Security OAuth), with a missing issuer what happens?

(IIRC Spring Security will fallback to protecting your application with Basic Auth and a generated password, but the application start up will NOT fail)

Reply:
Properties as "token-uri" and "authorization-uri" are mandatory for OAuth 2.0. Application will not start without them, it needs to get tokens from somewhere.
In our Okta App these properties depends on "**issuer-uri**"
https://github.com/okta/okta-spring-boot/blob/8b8a58ca5048abcc9f3b372583a4af4ac628b018/oauth2/src/main/java/com/okta/spring/boot/oauth/env/OktaOAuth2PropertiesMappingEnvironmentPostProcessor.java#L152
That's why without "issuer-uri" our app fails during configuration phase here:
https://github.com/spring-projects/spring-security/blob/48ef27b80ac449c3221d1da057ab2d803482e903/oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/registration/ClientRegistration.java#L673

Reply:
@bdemers I guess @VitaliiTytarenko-okta 's point here is that the standard Springboot Oauth requires `issuer` to be set to construct `token-uri` and `authorization-uri` without which the server would not start. So, for Oktaspringboot we should probably do a similar thing of failing to start the server when `okta.oauth2.issuer` is not set.

Reply:
I just created a new Spring Security project from start.spring.io, I added the two non-okta & non-cloud oauth options. added a simple rest controller, and started the application with no properties set

Spring Security falls back to Basic auth and generates a password:

```txt
Using generated security password: <password here>
```

Let chat about this, I feel like there is a fair bit of miss-communication (I want to make sure it's not me causing this 😉 )

Reply:
This is a pretty simple application without a husk, it has only `org.springframework.boot` dependencies and this is very close to what we have: https://github.com/oktadeveloper/okta-spring-boot-authz-server-example/tree/master/SpringBootOAuthClient
It fails without "token-uri" or "authorization-uri".
Full article: https://developer.okta.com/blog/2019/03/12/oauth2-spring-security-guide

Reply:
That application is expected to fail, due to this line: https://github.com/oktadeveloper/okta-spring-boot-authz-server-example/blob/master/SpringBootOAuthClient/src/main/java/com/okta/spring/SpringBootOAuthClient/SecurityConfiguration.java#L15

Which explicitly tries to configure OAuth.

The user could also programmatically configure the related OAuth bits (without properties), we need to make sure our integration does not break things.